### PR TITLE
Smart Contracts documentation (interpreter1)

### PR DESCRIPTION
**Pull request by bchamagne**
This is the documentation for https://github.com/archethic-foundation/archethic-node/pull/903
As discussed, please think about this PR as a wiki. Edit it directly!

### CODE REVIEW

File: docs/build/smart-contracts/reference/library.md [hunk 1]
```diff
@@ -0,0 +1,668 @@
...
+- `str` the text to hash
+- `algo` the hash algorithm ("sha256", "sha512", "sha3_256", "sha3_512", "blake2b")
+
+Returns the hash of `str` by using `algo` algorithm.
+
+---------
+
+## Token
+
+### hash/2
```
Comment (Neylix) [inline docs/build/smart-contracts/reference/library.md:511]: Wrong function name

File: docs/build/smart-contracts/reference/library.md [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+```
+
+Parameters:
+- `address` the token address
+
+Returns the token's id of the token at `address`.
+
+---------
+
+## Contract
```
Comment (Neylix) [inline docs/build/smart-contracts/reference/library.md:524]: Maybe contrat should be explained also in actions part to explain how to build the next transaction

File: docs/build/smart-contracts/reference/condition.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+1. If the expression return false, this "boolean expression" fails.
+1. If the expression return a value, this "boolean expression" passes if the transaction's property has the same value.
+
+:::info
+In these blocks, there is also some sugar to automatically add the property as an argument of the functions called. For example `uco_transfers: Map.size() > 0` will automatically expand to `uco_transfers: Map.size(transaction.uco_transfers) > 0`.
+:::
+
+## Condition inherit
+
+The `condition inherit` purpose is to check the next transaction generated by the smart contract after its code execution. It ensure the next transaction respect specific rules so the smart contract chain cannot be compromised.
```
Comment (cluster) [inline docs/build/smart-contracts/reference/condition.md:59]:
```suggestion
The `condition inherit` purpose is to check the next transaction generated by the smart contract after its code execution. It ensures the next transaction respects specific rules so the smart contract chain cannot be compromised.
```

File: docs/build/smart-contracts/reference/condition.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+:::
+
+There are 2 global variables for this condition block:
+1. `previous` is the transaction of the current contract.
+1. `next` is the "next transaction".
+
+See [Action's Appendix 1](/build/smart-contracts/reference/actions#appendix-1-the-transaction-map) for the details of the transaction map.
+
+:::caution
+A specific rule is applied for inherit condition, if a field is not specified in the conditions, it assume that it must have the same value as the previous transaction. ie. `code: previous.code == next.code`
```
Comment (cluster) [inline docs/build/smart-contracts/reference/condition.md:76]:
```suggestion
A specific rule is applied for inherit condition, if a field is not specified in the conditions, it assumes that it must have the same value as the previous transaction. ie. `code: previous.code == next.code`
```

File: docs/build/smart-contracts/reference/library.md [hunk 5]
```diff
@@ -0,0 +1,657 @@
...
+```elixir
+String.to_number("12")     # 12
+String.to_number("-12")    # -12
+String.to_number("12.1")   # 12.1
+```
+
+Parameters:
+- `str` the string
+
+Returns the number whose text representation is `str`.
```
Comment (cluster) [inline docs/build/smart-contracts/reference/library.md:65]:
```suggestion
Returns the number which text representation is `str`.
```

File: docs/build/smart-contracts/reference/library.md [hunk 6]
```diff
@@ -0,0 +1,657 @@
...
+```
+
+Parameters:
+- `text` the list
+- `pattern` the element
+
+Returns the data from `text` extract via the regex `pattern`. 
+
+:::note
+This function require the usage of capture groups (subexpressions).
```
Comment (cluster) [inline docs/build/smart-contracts/reference/library.md:343]:
```suggestion
This function requires the usage of capture groups (subexpressions).
```

File: docs/build/smart-contracts/reference/library.md [hunk 7]
```diff
@@ -0,0 +1,657 @@
...
+```
+Parameters:
+- `any` anything
+
+Returns the Json representation of `any`.
+
+### parse/1
+
+```elixir
+Json.parse("1")                     # "1"
```
Comment (cluster) [inline docs/build/smart-contracts/reference/library.md:396]: does it return `1` or `"1"`?

File: docs/build/smart-contracts/reference/triggers.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+sidebar_label: Triggers
+sidebar_position: 1
+---
+
+# Triggers
+### Contract triggered by a transaction
+
+:::tip
+To trigger a SC execution, the incoming transaction must add the SC address in the `recipients` part of the transaction.  
+This allow to activate a SC without sending funds to it, or to send funds to it without activating the code execution. As the `recipients` field is a list, you can trigger multiple contract in the same transaction
```
Comment (cluster) [inline docs/build/smart-contracts/reference/triggers.md:13]:
```suggestion
This allows to activate a SC without sending funds to it, or to send funds to it without activating the code execution. As the `recipients` field is a list, you can trigger multiple contracts in the same transaction
```

**Review by cluster — APPROVED** [review]
very nice! 👏 
I've fixed a few typos and left one question

**Pull request merged**